extra/resources/ifspeed

```diff
@@ -165,13 +182,36 @@ monitor() {
     return ${ret}
 }
 
+# This function tries to guess nic interface by IP
+get_nic_name_by_ip(){
+    # $FINDIF takes its parameters from the environment
+    NICINFO=`${FINDIF}`
+    rc=$?
+    if [ $rc -eq 0 ];then
+        NICINFO=`echo "${NICINFO}" | sed -e 's/netmask\ //;s/broadcast\ //'`
+        echo "${NICINFO}" | cut -d" " -f1
```

Comment:
These two lines can be changed to
echo "${NICINFO%% *}"
No need to run sed/cut.

Reply:
I'm not familiar with this syntax. I think it is more readable like that.
What do you think?

Reply:
http://www.tldp.org/LDP/abs/html/string-manipulation.html
${string%%substring}
Deletes longest match of $substring from back of $string.
substring is ' *' (space and then anything)
%% removes the longest match at the end of string

And, that is a way faster.

Reply:
Yeah, the shell string stuff is horribly unreadable ... as bad as Perl ;) but it has a couple of advantages: it eliminates a dependency on external commands, and it is more efficient (because the shell parses and executes it directly, rather than spawn multiple external processes that must manage everything via I/O). It can help to add a comment reminding readers what it accomplishes.

Reply:
Thanks for the detailed explanation. I will change that and add a descriptive comment.